Fix #2746 and Fix part of #1723: Update Onboarding Instructions and Bazel Documentation

adhiamboperes opened the pull request:
Fix #2746:

- Remove gradle specific instructions, reorganize setup order, add instructions for running app and tests.
- Audit existing Bazel setup instructions and expand where necessary.
-  Installing the Bazel plugin in Android Studio
-  Syncing (& partial syncing)

Fix part of #1723:
- Ensure the installation instructions provide support for folks both using emulators and physical Android devices

**Other changes**
- Removed steps for setting up a Rosetta terminal because that was only required for modifying Gradle's `local.properties` file.
- `noexperimental_check_desugar_deps` flag is nolonger required. In Bazel 4, this was required to disable `[--experimental_check_desugar_deps](https://docs.bazel.build/versions/master/command-line-reference.html#flag--experimental_check_desugar_deps), which had several incompatibilities with the Apple Silicon Chip. For more information, please see [here](https://blog.bazel.build/2018/12/19/bazel-0.21.html#android).

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
N/A

adhiamboperes committed:
Modify onboarding instructions

Remove gradle specific instructions, reorganize setup order, add instructions for running app and tests.

adhiamboperes committed:
Remove references to Rosetta for M1

adhiamboperes committed:
Update Bazel on Mac documentation

adhiamboperes committed:
Add documentation for running the app on a physical device and emulator.

adhiamboperes commented:
@manas-yu, could you PTAL at this PR? I have made some changes that need to be reviewed based on:
 - Are they clear, or do we need to paraphrase some sentences?
 - Are they easy to follow?
 - Are there any grammatical errors?
 The sections in red are deletions, and green are additions.

Ps - you can switch to rich diff on the top right of each file to see the MD formatted version.

@BenHenning, could you PTAL for completeness?

manas-yu commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -14,23 +14,48 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+   - Emulators provide the flexibility of testing the application on a variety of devices and Android API levels without needing to have each physical device.
+   
+     **Emulator system requirements**
+       For the best experience, you should use the emulator in Android Studio on a computer with at least the following specs:
```
The bullet points (if that was the intended format) below are not rendered correctly due to indentation.

adhiamboperes replied:
Fixed.

manas-yu commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -14,23 +14,48 @@ This wiki page explains how to install Oppia Android on your local machine. If y
-   **Direct download Url**: [Windows](https://redirector.gvt1.com/edgedl/android/studio/install/2021.1.1.23/android-studio-2021.1.1.23-windows.exe) | [Linux](https://redirector.gvt1.com/edgedl/android/studio/ide-zips/2021.1.1.23/android-studio-2021.1.1.23-linux.tar.gz) | [Intel Mac](https://redirector.gvt1.com/edgedl/android/studio/install/2021.1.1.23/android-studio-2021.1.1.23-mac.dmg) | [Apple  ...
-
-   You can have multiple versions of Android Studio installed on your machine at the same time.
+1. Download/Install the latest version of Android Studio from [this location](https://developer.android.com/studio).
```
Would it be reasonable to recommend the latest version even after encountering the syntax highlighting issue in Ladybug AS?

adhiamboperes replied:
I've added a note to recommend Giraffe, but we are not opposed to folks testing and reporting issues with newer versions.

manas-yu commented on wiki/Contributing-to-Oppia-Android.md:
```diff
@@ -59,15 +59,14 @@ Once you've merged PRs that correspond to **two** different pre-existing GitHub
-- That said, when submitting a PR for review, you may notice that some Bazel-specific tests or workflows fail. Investigating and fixing these will require setting up Bazel in your local environment (see the instructions [here](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions)), and then running the specific Bazel commands in your local repository (most team members just  ...
-- Bazel & Gradle sometimes don't play nicely with one another. So, when you're verifying Bazel-specific things, we recommend doing so in one go, and then deleting the corresponding Bazel build artifacts using ``bazel clean`` before switching back over to Gradle (to avoid any issues with the two build systems crossing). Note that Bazel generally doesn't have any problems with Gradle build artifact ...
-- As the team finishes the migration to Bazel, communications and instructions will be sent ahead of time for moving development environments away from Gradle so that we can officially deprecate it.
+  - Syntax highlighting errors in some instances.
```
If needed, it be more specific to mention the actual IDE issue: "Syntax highlighting has been temporarily turned off due to an internal error."

adhiamboperes replied:
Thanks. This wiki section has been removed in latest changes.

manas-yu commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +84,154 @@ Please follow these steps to set up Oppia Android on your local machine.
+
+   <img width="1029" alt="Image" src="https://github.com/user-attachments/assets/efcc4e6e-b6ba-44fc-9c96-aa381de83911" />
+
+3. A setup wizard will open. For Workspace, click (**...**) to open file manager and select /opensource/oppia-android as the location since this is where we have our project's `WORKSPACE` file.
```
```suggestion
3. A setup wizard will open. For Workspace, click (**...**) to open file manager and select `/opensource/oppia-android` as the location since this is where we have our project's `WORKSPACE` file.
```

adhiamboperes replied:
Done

manas-yu commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +84,154 @@ Please follow these steps to set up Oppia Android on your local machine.
+      .
+      
+   # Automatically includes all relevant targets under the 'directories' above
+   derive_targets_from_directories: true
```
In the [gist](https://gist.github.com/BenHenning/676113880e469727fe752c149da98e5b) by Ben, the property `derive_targets_from_directories` is set to `false`. I have also set it to `false` since we have already explicitly specified targets under the `targets` field below.  

setting it to true does not make much difference in the build but could lead to unexpected targets being indexed—just wanted to point it out for discussion.

adhiamboperes replied:
Noted and updated

BenHenning replied:
FWIW I would actually expect there to be a rather steep time and resource penalty for using derive targets unless there's some extra simplifying happening in the plugin (or if your custom targets are very extensive) since this will pull in *all* targets (including hundreds of tests). For me, it's the difference between an ~50 minute and ~100+ minute sync.

manas-yu commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +84,154 @@ Please follow these steps to set up Oppia Android on your local machine.
 
-### Configure Emulator Tests
+   - Test **name**, is helpful for identifying the test target from the list on the left.
+   - **Target expression** requires the fully qualified path to the test file.
```
Would it be better to add information on how to obtain the correct qualified path for the test after making changes to the actual test file path?

adhiamboperes replied:
I have added this above

manas-yu reviewed (commented):
Hi @adhiamboperes, PTAL. I’ve left some inline comments with my initial thoughts while reviewing the documentation. The changes look clear and well-structured, and I didn’t notice any grammatical errors.

BenHenning commented:
Thanks @adhiamboperes. Been trying, and failing, to get to this so I'm very happy to see that you were able to make some time to get this situation better.

I will take a look for content soon, but one thought: could we maybe consolidate all starting documentation to just a single location? It's currently confusing that it's so split up, and now that we're using Bazel the differences between the different platform setups seem less significant. Suggestions:
- Revamp the installation page to cover both app install & Bazel setup instructions. Any platform-specific stuff should be added as subsections there (but keep common stuff together so we don't need to update copies of information).
- Consolidate installation issues to the installation page rather than having a separate one.
- Move known Bazel issues to the installation issues list from the contributing page.
- Remove the contributing page's sections on Bazel migration and installing the Oppia web app (we don't need the latter).
- Please make sure there aren't references to the old installation pages anywhere in code or the wiki pages. It'd be nice to do a redirect, but as far as I can tell that's not possible for wiki pages.

WDYT?

Edit: Apparently the contributing page _can_ be found, it's just called 'Getting started' in the side panel. I hadn't realized that earlier. :)

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into update-documentation

adhiamboperes committed:
Re-organize wiki.

adhiamboperes reviewed (commented):
Thanks @manas-yu! I have addressed all your comments.
 
@BenHenning, 
> * Revamp the installation page to cover both app install & Bazel setup instructions. Any platform-specific stuff should be added as subsections there (but keep common stuff together so we don't need to update copies of information).

Done

> * Consolidate installation issues to the installation page rather than having a separate one.

Done, and deleted previous  issues page.

> * Move known Bazel issues to the installation issues list from the contributing page.

Done.

> * Remove the contributing page's sections on Bazel migration and installing the Oppia web app (we don't need the latter).

Done

> * Please make sure there aren't references to the old installation pages anywhere in code or the wiki pages. It'd be nice to do a redirect, but as far as I can tell that's not possible for wiki pages.

I have have done the cleanup.

> WDYT?

Generally, this is a great idea. Reducing the need to navigate back and forth has always been a goal -- we started this in https://github.com/oppia/oppia-android/issues/5099.
 
@BenHenning, PTAL.

adhiamboperes committed:
Fix formatting

BenHenning commented on wiki/Troubleshooting-Installation.md:
```diff
@@ -1,128 +0,0 @@
-
-   To fix this error, you need to lower the version of Java to compile the JAR file. Please see [here](https://developer.android.com/studio/intro/studio-config#jdk) for more information about Java versions.
-
-### Bazel issues
```
Are these not worth keeping in the new documentation?

adhiamboperes replied:
I moved these over, except for the 
```
external/bazel_tools/src/tools/android/java/com/google/devtools/build/android/dexer/DexFileSplitter.java:21: error: package com.android.dex does not exist
   import com.android.dex.DexFormat;
```

which may have been useful during Bazel 4-6 migration, but is not likely to be an issue now.

BenHenning commented on wiki/Frequent-Errors-and-Solutions.md:
```diff
@@ -1,33 +0,0 @@
-## Table of Contents
```
Are these errors not worth keeping in the new documentation?

adhiamboperes replied:
I felt that the tablet layout error was pretty specific and not necessarily a frequently encountered error. However, I have moved it to the new errors section. The other errors are covered too.

BenHenning commented on wiki/Upgrading-Target-Sdk-Guide.md:
```diff
@@ -64,7 +64,7 @@ The new comment should include an explicit indication of whether the audit was c
 
 Compatibility with the new target SDK should be done by:
-1. Building a local production [Bazel build](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions#building-the-app) of the app (``//:oppia_beta``) and deployed to a local emulator or device running the **same** version of Android as the new target SDK version.
+1. Building a local production [Bazel build](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Glossary#building-the-app) of the app (``//:oppia_beta``) and deployed to a local emulator or device running the **same** version of Android as the new target SDK version.
```
Might it be better to reference the installing documentation for this, instead?

adhiamboperes replied:
Done

BenHenning commented on wiki/Oppia-Bazel-Glossary.md:
```diff
@@ -3,60 +3,53 @@
-   - [For Linux](https://github.com/oppia/oppia-android/wiki/Bazel-Setup-Instructions-for-Linux)
+Follow the instructions on the [installation page](https://github.com/oppia/oppia-android/wiki/Installing-Oppia-Android#install-bazel) to set up Bazel.
 
 ### Building the app
```
This probably should just reference the installation page, instead (or vice versa) to avoid duplicating the mobile-install instructions and caveat.

adhiamboperes replied:
Moved the CLI instructions here from the installation page.

BenHenning commented on wiki/Oppia-Bazel-Glossary.md:
```diff
@@ -3,60 +3,53 @@
 
 **Move your command line head to the `~/opensource/oppia-android`**, then run the below bazel command:
 
+On Sdk 29 and below, run:
```
```suggestion
On SDK 29 and below, run:
```

Nit.

adhiamboperes replied:
Fixed

BenHenning commented on wiki/Oppia-Bazel-Glossary.md:
```diff
@@ -133,3 +126,59 @@ oppia_android_test(
+
+**Partially Sync File with Bazel**
+Syncs only the targets corresponding to the file from which this action is invoked. Can cut down on the time to sync the project if you’re only interested in resolving a single file.
+- This is invoked from the current active file.
```
Perhaps mention as separate points that:
- This can be accessed via a right-click (for Windows/Linux).
- This is a great way to either sync a file which is currently annotated as '(Unsynced)' in the IDE or for ensuring API changes are seen by other targets.
- This can be used on directories which is especially useful for resyncing multiple libraries or regenerating protos.

adhiamboperes replied:
Added these points.

BenHenning commented on wiki/Oppia-Bazel-Glossary.md:
```diff
@@ -133,3 +126,59 @@ oppia_android_test(
+Syncs only the targets corresponding to the file from which this action is invoked. Can cut down on the time to sync the project if you’re only interested in resolving a single file.
+- This is invoked from the current active file.
+
+**Automatic Sync**
```
I've never used this setting--is it too slow for us? Wondering if we should recommend against using it for now (it currently reads as "it might work, we don't know").

adhiamboperes replied:
It is most likely not performant. I have added the recommendation.

BenHenning commented on wiki/Oppia-Bazel-Glossary.md:
```diff
@@ -3,60 +3,53 @@
 ## Overview
 Bazel is an open-source build and test tool similar to Make, Maven, and Gradle. It uses a human-readable, high-level build language.
 
 **WARNING: We recommend to not use the Android Studio Bazel plugin since it currently has compatibility issues with the project.**
```
Drop this warning as it's directly counter to the guidance introduced in this PR.

adhiamboperes replied:
Done.

BenHenning commented on wiki/Oppia-Bazel-Glossary.md:
```diff
@@ -3,60 +3,53 @@
 - [Overview](#overview)
```
Perhaps this page should be 'Bazel user guide' rather than 'glossary'? It's not defining terms, it seems more for outlining "things you should know to be efficient" which feels more like a user guide. WDYT?

adhiamboperes replied:
Thanks - I was not sure about that naming either.

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,313 @@ This wiki page explains how to install Oppia Android on your local machine. If y
-
-   You can have multiple versions of Android Studio installed on your machine at the same time.
+1. Download/Install the latest version of Android Studio from [this location](https://developer.android.com/studio).
+   **Note**: We strongly recommend **Giraffe** because it has been tested and produced the best experience across different OSes.
```
Suggest flipping this:

"Download and install Android Studio Giraffe from ...
**Note**: You may try newer or older versions of Android Studio, but we can't guarantee good support."

Otherwise we're forcing the reader to make a decision they may not have the context to reasonably make. Separately, we may need to link to the archive rather than the main AS page since it can be hard to find old versions.

adhiamboperes replied:
Updated.

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,313 @@ This wiki page explains how to install Oppia Android on your local machine. If y
-   - In Android Studio, open Tools > SDK Manager.
-     - In the "SDK Platforms" tab (which is the default), select `API Level 28` and also `API Level 30` (for Bazel support).
+   - In Android Studio, open **Tools** > **SDK Manager**.
+     - In the "SDK Platforms" tab (which is the default), select `API Level 29` and also `API Level 30`.
```
Should be 30 and 34 I think. Robolectric still (probably) uses 30 in Bazel, but we're targeting 34 now for production code.

adhiamboperes replied:
Done

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,313 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+
+       - In Android Studio, in the right hand toolbar, locate `device Manager`.
+       - Click the **+**, and then click **Create Virtual Device**.
+       - Create a phone and a tablet AVD.
```
Probably need to specify the SDK to use. Perhaps we can recommend to always install the SDK version listed above for installation?

adhiamboperes replied:
Added this information.

BenHenning commented on wiki/Installing-Oppia-Android.md:
````diff
@@ -5,32 +5,313 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+
+Run:
+```shell
+sudo apt update && sudo apt install bazel-6.5.0
````
```suggestion
sudo apt-get update && sudo apt-get install bazel-6.5.0
```

`apt` isn't a stable CLI API (which it likes to report :)) and this is when `apt-get` should be used.

adhiamboperes replied:
Ohh...Updated.

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,313 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+
+**Step 3: Set up your environment**
+
+Ensure that your `ANDROID_HOME` environment variable is set to the location of your Android SDK. To do this, find the path to the installed SDK using Android Studio’s SDK Manager (install SDK 28). Assuming the SDK is installed to default locations, you can use the following commands to set the `ANDROID_HOME` variable:<br>
```
Strike 'install SDK 28' here as it contradicts the installation above. We should provide instructions in one location for setting up the SDK, no need to repeat it.

Also, can the `<br>` be removed? Not sure why we need it here.

adhiamboperes replied:
Done.

BenHenning commented on wiki/Installing-Oppia-Android.md:
````diff
@@ -5,32 +5,313 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+```
+
+### Bazel Set up for Windows
+This page outlines one way to allow Bazel to be used in CLI form on Windows. Please note that **this support is currently experimental**. You may run into some problems--we suggest that you post a discussion at [github-discussions](https://github.com/oppia/oppia-android/discussions).
````
```suggestion
This page outlines one way to allow Bazel to be used in CLI form on Windows. Please note that **this support is currently experimental**. We suggest that you post a discussion at [github-discussions](https://github.com/oppia/oppia-android/discussions) if you run into any problems.
```

adhiamboperes replied:
Done.

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,313 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+
+Unlike Unix-based systems where Bazel runs natively without issue, the current solution on Windows is to install an Ubuntu-based subsystem. Windows currently only supports a terminal experience in this subsystem (though there is a prerelease version of the software with GUI support) which means Android Studio will not be supported. You will need to continue using the Windows version of Android St ...
+
+Due to the issues mentioned above, we recommend dual-booting your PC with Linux. However, if you choose to use windows, please follow the instructions below:
```
```suggestion
Due to the issues mentioned above, we recommend dual-booting your PC with Linux. However, if you choose to use Windows, please follow the instructions below:
```

adhiamboperes replied:
Fixed

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,313 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+Due to the issues mentioned above, we recommend dual-booting your PC with Linux. However, if you choose to use windows, please follow the instructions below:
+
+**Main prerequisites**:
+- Windows 10: These instructions are geared towards users of Windows 10 (older versions will not be compatible). If you're in using an older version of Windows, please follow up with a comment on [this issue](https://github.com/oppia/oppia-android/issues/3371).
```
@Rd4dev can we say 'Windows 11' here, instead? Windows 10 is no longer supported by MS and we should probably refer to the correct thing (assuming it actually works with the same instructions).

Ditto for the Windows 10 reference below.

adhiamboperes replied:
Updated to 10+ for now

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,313 @@ This wiki page explains how to install Oppia Android on your local machine. If y
 2. Configure your Android Studio
-   - In Android Studio, open Tools > SDK Manager.
-     - In the "SDK Platforms" tab (which is the default), select `API Level 28` and also `API Level 30` (for Bazel support).
+   - In Android Studio, open **Tools** > **SDK Manager**.
```
These instructions should be framed before opening the project--are they still understandable in that context? Android Studio's out of box experience can be a bit confusing at times.

adhiamboperes replied:
Updated to setup the tools from the welcome screen.

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
-   - Then, click "Apply" and "OK" to complete your setup.
-5. Click the elephant icon in the toolbar ("Sync Gradle") to ensure that all the correct dependencies are downloaded. (In general, you'll want to do this step any time you update your dependencies.)
-
 ## Run the app from Android Studio
```
Should this be 'opening the project'?

adhiamboperes replied:
Updated to "Opening the project in Android Studio"

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
+
+   <img width="760" alt="Image" src="https://github.com/user-attachments/assets/5a26210f-865d-4048-b5a5-73379984bfb5" />
+
+7. In the above file, modify it so that it looks like below(copy and paste):
```
```suggestion
7. In the above file, modify it so that it looks like below (copy and paste):
```

Nit on spacing.

adhiamboperes replied:
Fixed

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
+   
+   targets:
+   # If source code isn't resolving, add additional targets that compile it here
+      //:oppia_dev_binary
+      //app
+      //app/src/main/...
+      //domain/...
+      //model/...
+      //testing/...
+      //utility/...
+      //data/...
+      //scripts/...
```
Probably too extensive. Let's do:

```
      //:oppia_dev_binary
      //app
      //app/src/main/...
      //domain/src/main/...
      //testing/src/main/...
      //utility/src/main/...
      //data/src/main/...
      //model/...
```

This drops scripts and all tests for a much faster sync.

adhiamboperes replied:
Updated

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
+
+   # Please uncomment an android-SDK platform. Available SDKs are:
+      # android_sdk_platform: android-28
+      android_sdk_platform: android-29
```
Should be 30.

adhiamboperes replied:
Updated

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
+      android_sdk_platform: android-34
+   
+   shard_sync: true
+   sync_flags:
+      --local_ram_resources=HOST_RAM*.5
+      --discard_analysis_cache
```
Drop this--it may be useful to bring back in the future, but I don't think it's needed with Bazel 6.x. Plus, discarding the analysis cache will really slow down even incremental syncs for less performant computers.

adhiamboperes replied:
Dropped

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
+
+9. Once sync has finished, you can now build and install the app on either a virtual or physical device. Bazel supports deploying to only one device at a time, so you can connect one device, or launch one emulator at a time.
+
+   You can run the project by using the Bazel plugin to set up run configurations for the target that you wish to build. This performs the same action as the commands below, but using the GUI to run the app might be more intuitive for some developers.
```
FWIW we can actually save these configurations for people to use, too:

https://ij.bazel.build/docs/run-configurations.html#sharing-run-configurations

Perhaps that's worth doing and updating the project view above to import it?

adhiamboperes replied:
Should be solved when https://github.com/oppia/oppia-android/issues/5761 is addressed

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
+     - **Bazel flags** are optional.
+     - Select **Apply** and then **Close** or **Ok**.
+
+## Run the app from the command line
```
As mentioned in another comment, but perhaps this should go in the user guide and this updated to point to that?

In fact, I wonder if a lot of the "here's important Bazel stuff you need to do often or understand" should go in the user guide as a basis, and this page can be kept more for setting everything up.

adhiamboperes replied:
Moved to the user guide and included the link above.

BenHenning commented on wiki/Installing-Oppia-Android.md:
````diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
+```
 
-#### For tests that are in non-app modules, such as **domain** or **utility**::
+### Configure Robolectric Tests
````
Probably also better to put in the user guide?

adhiamboperes replied:
The whole setting up tests section?

adhiamboperes replied:
I kept the basic test instructions here and moved the rest of the instructions and tips to the user guide.

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +341,183 @@ Please follow these steps to set up Oppia Android on your local machine.
 
-3. In the test file, to the left of the class name, click on the orange and green arrow, and select **Run 'HomeActivityTest'**.
-   ![](https://user-images.githubusercontent.com/59600948/272657131-96e5354b-13a9-4709-969a-b9494a65c30f.png)
+#### For tests that are in non-app modules, such as **domain** or **utility**::
```
Others:
- Can right click on the tab if the test is open and copy the BUILD target.
- For all tests (including app), you can run 'bazel query' on the relative source path of the file (also copyable via right click on the file tab when the test is open) and remove the '.kt' at the end of the returned value to get the test target.

adhiamboperes replied:
These are covered under `Set up and run tests` tips above.

adhiamboperes replied:
Transferred to teh to the user guide.

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -150,8 +526,95 @@ We strongly recommend running tests on Robolectric which is faster because it do
+The team is in the process of migrating away from Gradle to Bazel, so while the project still has Gradle build files, they are no longer maintained and cannot be used to build the project.
+
+Please note that:
+- It's currently recommended that all team members use **Bazel** for their active development in Android Studio. While some team members use the Bazel Android Studio plugin, day-to-day Bazel development currently suffers from:
```
```suggestion
- The IntelliJ Bazel plugin currently has some known issues:
```

adhiamboperes replied:
Updated

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -150,8 +526,95 @@ We strongly recommend running tests on Robolectric which is faster because it do
+    - Syntax highlighting errors in some instances.
+- As the team finishes the migration to Bazel, communications and instructions will be sent ahead of time, and we will continually update our documentation with the latest changes.
+
+1. No matching toolchains (sdk_toolchain_type)
```
```suggestion

Other problems that you may run into:

1. No matching toolchains (sdk_toolchain_type)
```

(to provide some spacing between the 'please note that' and the numeration of problems).

adhiamboperes replied:
Done

BenHenning reviewed (changes requested):
Thanks so much @adhiamboperes! This is a really nice move toward consolidating and simplifying setup and onboarding.

I have a lot of comments, but the majority are aiming at reducing potential confusion pathways and encourage further simplification and/or consolidation. PTAL.

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into update-documentation

adhiamboperes committed:
Address reviewer comments.

adhiamboperes committed:
Fix toc check failures

adhiamboperes commented:
PTAL @BenHenning. Could you also please clarify https://github.com/oppia/oppia-android/pull/5723#discussion_r1994476963?

Also, something happened when I was renaming the "Glossary" page, did not use `git mv `and instead it appears to be deleted and "User Guide" is a replacement.

manas-yu commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,221 @@
+## Concepts and Terminology
+**[Workspace](https://github.com/oppia/oppia-android/blob/develop/WORKSPACE)**
+
+A workspace is a directory where we add targeted SDK version, all the required dependencies and there required Rules. The directory containing the WORKSPACE file is the root of the main repository, which in our case is the `oppia-android` root directory is the main directory.
```
```suggestion
A workspace is a directory where we specify the targeted SDK version, required dependencies, and corresponding Rules. The directory containing the WORKSPACE file is the root of the main repository, which in our case, is the `oppia-android` root directory.
```

adhiamboperes replied:
Done

manas-yu commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +350,145 @@ Please follow these steps to set up Oppia Android on your local machine.
+   targets:
+   # If source code isn't resolving, add additional targets that compile it here
+      //:oppia_dev_binary
+      //app //app/src/main/... 
```
```suggestion
      //app 
      //app/src/main/... 
```

adhiamboperes replied:
Fixed

manas-yu commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -63,85 +350,145 @@ Please follow these steps to set up Oppia Android on your local machine.
-- This configuration will run all the tests in that class.
-5. To run only a specific test in a file:
-   - Search or scroll down to the desired test name, to the left of the test name, click on the run icon and select **Run '`test name`''**.
+3. The third way is to run a the test via CLI as mentioned above.
```
```suggestion
3. The third way is to run the test via CLI as mentioned above.
```
We could also suggest here, the test files that are not yet modularized (i.e., test packages currently missing BUILD files) need to be executed via the CLI.

adhiamboperes replied:
Included this info, but can now be found in the user guide.

manas-yu reviewed (commented):
Thanks @adhiamboperes PTAL. The changes look good to me—just wanted to highlight a few things.

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into update-documentation

adhiamboperes committed:
Further docs reorg

adhiamboperes commented:
Thanks for the follow up @manas-yu! I have made the changes.

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,322 @@ This wiki page explains how to install Oppia Android on your local machine. If y
 ## Table of Contents
 
 - [Prepare developer environment](#prepare-developer-environment)
+- [install Bazel](#install-bazel)
```
```suggestion
- [Install Bazel](#install-bazel)
```

adhiamboperes replied:
Done.

BenHenning commented on wiki/Installing-Oppia-Android.md:
````diff
@@ -365,60 +374,55 @@ Please follow these steps to set up Oppia Android on your local machine.
-   java_language_level: 11
-   ```
+```
+directories:
````
Let's replace this with an `import` statement per #5762 and specifically https://ij.bazel.build/docs/project-views.html#import. I think that will look like this:

```
import oppia-android.bazelproject
```

adhiamboperes replied:
Done. Also modified the import instructions.

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -526,23 +481,33 @@ bazel test //domain/src/test/java/org/oppia/android/domain/onboarding:AppStartup
 - To learn about how we write tests at oppia-android, please refer to: [Oppia Android Testing](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing).
 
-- To learn more about getting started with Bazel, please see [Oppia-Bazel-Glossary](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Glossary).
+- To learn more about getting started with Bazel, please see the [Bazel User Guide](https://github.com/oppia/oppia-android/wiki/Bazel-User-Guide).
```
```suggestion
- An important reference to go back to as you continue developing with Bazel is the [Bazel User Guide](https://github.com/oppia/oppia-android/wiki/Bazel-User-Guide).
```

'Getting started' sounds a bit too much like 'start here' which isn't correct--_this_ page is the actual starting place and the guide is just a long-term reference.

adhiamboperes replied:
Done.

BenHenning commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,277 @@
+  - [Common Flags](#common-flags)
+  - [Running multiple test targets](#running-multiple-test-targets)
+- [Concepts and Terminology](#concepts-and-terminology)
+- [Syncing the Project](#Syncing-the-project)
```
This doesn't seem linkified correctly--please fix.

adhiamboperes replied:
Fixed.

BenHenning commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,277 @@
+`--cache_test_results=no` → Force re-running tests even if cached.
+`--test_filter=ClassName#methodName` → Run specific test cases.
+
+### Running multiple test targets
```
Another example to add: you can append multiple test targets to the same line, e.g.:

```sh
bazel test -- //path/to/target/OneTest //path/to/target/TwoTest
```

adhiamboperes replied:
Added

BenHenning commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,277 @@
+A rule specifies the relationship between inputs and outputs, and the steps to build the outputs.
+In Android, rules are defined using `android_binary`. Android rules for testing are `android_instrumentation_test` and `android_local_test`.
+
+**[BUILD files](https://github.com/oppia/oppia-android/blob/develop/app/BUILD.bazel)**
```
Please permalink to this & the WORKSPACE files, too. Even these files aren't permanent and will probably disappear one day (app's eventually with modularization maturity and WORKSPACE when we migrate to Bazel 8.x).

adhiamboperes replied:
Done

BenHenning commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,277 @@
+   * The output will be a Bazel target that starts with //. Copy the target and remove the .kt extension.
+   * Use this target in the bazel test command, e.g. `bazel test //domain/src/test/java/org/oppia/android/domain/onboarding:AppStartupStateControllerTest`.
+
+### Common Flags
```
Another good flag to add: `--runs_per_test` to test multiple runs: https://bazel.build/reference/command-line-reference#flag--runs_per_test.

adhiamboperes replied:
Added

BenHenning commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,277 @@
+ - `visibility = ["//visibility:public"],` - Anyone can use this target.
+ - `"//visibility:private"` - Only targets in this package can use this target.
+
+**[Testing](https://github.com/oppia/oppia-android/blob/ba8d914480251e4a8543feb63a93b6c91e0a5a2f/app/BUILD.bazel#L719)**
```
Can we update this to point to an oppia_android_test now that app_test was removed? I missed it in my PR since there wasn't an easy way to find this reference.

adhiamboperes replied:
Done

BenHenning commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,277 @@
+load("//:oppia_android_test.bzl", "oppia_android_test")
+
+oppia_android_test(
+    name = "MathExpressionAccessibilityUtilTest",
+    srcs = ["MathExpressionAccessibilityUtilTest.kt"],
+    custom_package = "org.oppia.android.app.utility.math",
+    test_class = "org.oppia.android.app.utility.math.MathExpressionAccessibilityUtilTest",
+    test_manifest = "//app:test_manifest",
+    deps = [
+      ...
+    ],
```
Can we make sure the order here is consistent between this block and the one above? Also, they should follow whatever buildifier enforces.

adhiamboperes replied:
Done

BenHenning commented on wiki/Bazel-User-Guide.md:
````diff
@@ -0,0 +1,277 @@
+   ```
+   adb install bazel-bin/oppia_dev_binary.apk  
+   ```
+* Starting from Sdk 30, incremental builds, like those executed using `bazel mobile-install`, are no longer permitted, necessitating the use of two separate commands.
````
Another case for `SDK`. Perhaps do a quick search across the wiki to ensure consistency?

adhiamboperes replied:
Fixed. This is the only occurance found.

BenHenning commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,322 @@ This wiki page explains how to install Oppia Android on your local machine. If y
 
-   **Direct download Url**: [Windows](https://redirector.gvt1.com/edgedl/android/studio/install/2021.1.1.23/android-studio-2021.1.1.23-windows.exe) | [Linux](https://redirector.gvt1.com/edgedl/android/studio/ide-zips/2021.1.1.23/android-studio-2021.1.1.23-linux.tar.gz) | [Intel Mac](https://redirector.gvt1.com/edgedl/android/studio/install/2021.1.1.23/android-studio-2021.1.1.23-mac.dmg) | [Apple  ...
+   - Then, click "Apply" to download and install these two SDKs/Tools.
+     - On this screen, note the **Android SDK Location**, which is set as the ANDROID_HOME variable in  the Bazel setup step below.
```
```suggestion
     - On this screen, note the **Android SDK Location**, which is set as the ANDROID_HOME variable in the Bazel setup step below.
```

Spotted an extra space.

adhiamboperes replied:
Fixed.

BenHenning reviewed (commented):
Thanks @adhiamboperes. This looks really good!! I have very few concerns, just some small nits and content updates.

It'd be good to get some others' thoughts here beyond mine and @manas-yu. Perhaps @TanishMoral11 and @JishnuGoyal may have some thoughts (per the CLaM meeting this week).

adhiamboperes committed:
Address reviewer comments and refactor project view instructions.

adhiamboperes commented:
@BenHenning, PTAL.

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into update-documentation

BenHenning commented on wiki/Bazel-User-Guide.md:
```diff
@@ -110,6 +110,7 @@ Using the explicit Bazel target (// syntax) is more reliable compared to the fil
 `--test_output=all` → Show full test output.
 `--cache_test_results=no` → Force re-running tests even if cached.
 `--test_filter=ClassName#methodName` → Run specific test cases.
+`--runs_per_test=1000` → Specifies number of times to run each test.
```
```suggestion
`--runs_per_test=100` → Specifies number of times to run each test. This can be useful for checking whether your test is flaky.
```

I know it's just an example, but I suspect the vast majority of people should probably not even default to 1000 runs for any test, and people have a tendency to copy examples as-is before fulling understanding the nuance behind the choices. :)

I'm also not particularly caring of '100' just that something smaller than '1000' should probably be used, and this seemed like a reasonable number. I originally thought '10' but in reality this should be used for flake detection, and anything less than '100' is not going to give good odds for infrequent flakes (and we do use '1000' for the tests that need to be the most stable, such as the data provider tests, but I tend to run those myself since I have access to some good parallelization for cases like this).

adhiamboperes replied:
Done

BenHenning commented on wiki/Installing-Oppia-Android.md:
````diff
@@ -366,61 +366,31 @@ Please follow these steps to set up Oppia Android on your local machine.
+
+In the `.aswb` directory, you will find the generated `.bazelproject` file. It should contain:
+
+```shell
+import oppia-android.bazelproject
+```
````
Ooh, so importing a project view actually just generates the 'import' statement on its own? I want to double check this before approving since I expected that it would copy it.

adhiamboperes replied:
I tested the import, and this was the outcome.

BenHenning reviewed (commented):
Thanks @adhiamboperes! Just two follow-ups, PTAL.

adhiamboperes committed:
change example runs per test to 100

BenHenning reviewed (approved):
Thanks @adhiamboperes! Really excellent changes and simplifications here. I think we can go ahead and merge this when you're ready, but feel free to get some others' thoughts first. I think it may be better just to check this in, though, and adjust once more issues are brought up.

subhajitxyz commented on wiki/Bazel-User-Guide.md:
````diff
@@ -0,0 +1,284 @@
+load("@io_bazel_rules_kotlin//kotlin:android.bzl", "kt_android_library")
+```
+Here, we are loading `android.bzl` and we are going to use it with a symbol name `kt_android_library`.
+Arguments to the load function must be string literals. load statements must appear at top-level in the file.
````
load should start with a capital letter (Load).

subhajitxyz commented on wiki/Bazel-User-Guide.md:
````diff
@@ -0,0 +1,284 @@
+```
+deps = [ "//app",]
+```
+Here, `deps` is used to define the dependencies which is a type of dependencies called `deps dependencies` and it includes the files/directory/target which are dependent. From the above example the dependency is the `app` target which is defined in the [Build file of app package](https://github.com/oppia/oppia-android/blob/ba8d914480251e4a8543feb63a93b6c91e0a5a2f/app/BUILD.bazel#L616).
````
Missing a comma.
```suggestion
Here, `deps` is used to define the dependencies which is a type of dependencies called `deps dependencies` and it includes the files/directory/target which are dependent. From the above example, the dependency is the `app` target which is defined in the [Build file of app package](https://github.com/oppia/oppia-android/blob/ba8d914480251e4a8543feb63a93b6c91e0a5a2f/app/BUILD.bazel#L616).
```

adhiamboperes replied:
Done

subhajitxyz commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,284 @@
+
+**[Testing](https://github.com/oppia/oppia-android/blob/7344270032ac242b1b8987f1b51c8b5aa4f14ce3/app/src/sharedTest/java/org/oppia/android/app/player/exploration/BUILD.bazel#L7)**
+
+when we want to run test cases on Bazel build environment, a test target needs to be set up correctly:
```
Sentence should start with capital letter.
```suggestion
When we want to run test cases on Bazel build environment, a test target needs to be set up correctly:
```

adhiamboperes replied:
Done

subhajitxyz commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,284 @@
+
+**Expand Sync to Working Set**
+
+Your working set is any files your VCS says are dirty, roughly corresponding to something like git status. By default the plugin tries to expand the sync to cover any target in your working set. This ensures these files are refreshed without having to go to the trouble of adding a temporary target to your project view.
```
Missing a comma.
```suggestion
Your working set is any files your VCS says are dirty, roughly corresponding to something like git status. By default, the plugin tries to expand the sync to cover any target in your working set. This ensures these files are refreshed without having to go to the trouble of adding a temporary target to your project view.
```

adhiamboperes replied:
Done

subhajitxyz commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,322 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+     
+        <img width="1029" alt="Image" src="https://github.com/user-attachments/assets/efcc4e6e-b6ba-44fc-9c96-aa381de83911" />
+         
+        * If you already have a project open in Android Studio, in the right hand toolbar, locate `device Manager`.
```
```suggestion
        * If you already have a project open in Android Studio, in the right hand toolbar, locate `Device Manager`.
```

adhiamboperes replied:
Done

subhajitxyz commented on wiki/Installing-Oppia-Android.md:
````diff
@@ -5,32 +5,322 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+```
+
+### Bazel Set up for Mac
+It is recommended to use the binary installer steps outline below. For other ways to install Bazel on Mac, refer to the [official page](https://bazel.build/install/os-x).
````
```suggestion
It is recommended to use the binary installer steps outlined below. For other ways to install Bazel on Mac, refer to the [official page](https://bazel.build/install/os-x).
```

adhiamboperes replied:
Done

subhajitxyz commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,322 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+Due to the issues mentioned above, we recommend dual-booting your PC with Linux. However, if you choose to use Windows, please follow the instructions below:
+
+**Main prerequisites**:
+- Windows 10+: These instructions are geared towards users of Windows 10+ (older versions will not be compatible). If you're in using an older version of Windows, please follow up with a comment on [this issue](https://github.com/oppia/oppia-android/issues/3371).
```
```suggestion
- Windows 10+: These instructions are geared towards users of Windows 10+ (older versions will not be compatible). If you're using an older version of Windows, please follow up with a comment on [this issue](https://github.com/oppia/oppia-android/issues/3371).
```

adhiamboperes replied:
Done

subhajitxyz commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -150,8 +451,105 @@ We strongly recommend running tests on Robolectric which is faster because it do
+
+### Troubleshooting Installation
+#### General issues
+1. If the Static Checks on your PR keeps failing, please refer to [instructions on running static checks locally](https://github.com/oppia/oppia-android/wiki/Static-Analysis-Checks#how-to-run-static-checks-locally) in order to catch these errors before pushing.
```
‘keeps’ should be ‘keep’ to match the plural subject.
```suggestion
1. If the Static Checks on your PR keep failing, please refer to [instructions on running static checks locally](https://github.com/oppia/oppia-android/wiki/Static-Analysis-Checks#how-to-run-static-checks-locally) in order to catch these errors before pushing.
```

adhiamboperes replied:
Done.

subhajitxyz commented on wiki/Static-Analysis-Checks.md:
```diff
@@ -264,7 +264,7 @@ To fix failing tests from GitHub CI individually, follow the steps below.
 - You can also go to scripts/static_checks.sh to view the failing check and run it locally.
 
-Note: Before running the script command in your local terminal, make sure you have Bazel installed. To learn how to set up Bazel for Oppia Android, follow these [instructions](https://github.com/oppia/oppia-android/wiki/Oppia-Bazel-Setup-Instructions). Also make sure you have oppia-android-tools installed since static checks rely on these tools to be able to perform some of the checks. To install ...
+Note: Before running the script command in your local terminal, make sure you have Bazel installed. To learn how to set up Bazel for Oppia Android, follow these [instructions](https://github.com/oppia/oppia-android/wiki/Installing-Oppia-Android#install-bazel). Also make sure you have oppia-android-tools installed since static checks rely on these tools to be able to perform some of the checks. To ...
```
Missing a comma.
```suggestion
Note: Before running the script command in your local terminal, make sure you have Bazel installed. To learn how to set up Bazel for Oppia Android, follow these [instructions](https://github.com/oppia/oppia-android/wiki/Installing-Oppia-Android#install-bazel). Also, make sure you have oppia-android-tools installed since static checks rely on these tools to be able to perform some of the checks. To install oppia-android-tools, run `bash scripts/setup.sh` in the oppia-android directory.
```

adhiamboperes replied:
Done

subhajitxyz commented on wiki/Bazel-User-Guide.md:
````diff
@@ -0,0 +1,284 @@
+load("@io_bazel_rules_kotlin//kotlin:android.bzl", "kt_android_library")
+```
+Here, we are loading `android.bzl` and we are going to use it with a symbol name `kt_android_library`.
+Arguments to the load function must be string literals. load statements must appear at top-level in the file.
````
'load' should start with capital letter.
```suggestion
Arguments to the load function must be string literals. Load statements must appear at top-level in the file.
```

adhiamboperes replied:
Done

theayushyadav11 commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,284 @@
+
+**[Dependencies](https://github.com/oppia/oppia-android/blob/ba8d914480251e4a8543feb63a93b6c91e0a5a2f/BUILD.bazel#L16)**
+
+A target A depends upon a target B if B is needed by A at build. `A -> B`
```
```suggestion
A target A depends upon a target B, if B is needed by A at build. `A -> B`
```
A comma will make it more readable.

adhiamboperes replied:
Done

theayushyadav11 commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,284 @@
+   * Paste the target to the bazel test command, e.g. `bazel test //domain/src/test/java/org/oppia/android/domain/onboarding:AppStartupStateControllerTest`
+
+2. If the test file is open in Android Studio, but **Copy BUILD target string** is not available in the context menu:
+Sometimes the **Copy BUILD target string** is not available in the context menu either due to sync issues or modularization issues. You can still copy the file path, and pass it to the test command.
```
```suggestion
Sometimes, the **Copy BUILD target string** is not available in the context menu either due to sync issues or modularization issues. You can still copy the file path, and pass it to the test command.
```

adhiamboperes replied:
Done

TanishMoral11 commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,322 @@ This wiki page explains how to install Oppia Android on your local machine. If y
 
-1. Download/Install [Android Studio Bumblebee | Patch 3](https://developer.android.com/studio/archive).
+1. Download and install Android Studio **Giraffe** from [this location](https://developer.android.com/studio/archive).
+   **Note**: You may try newer or older versions of Android Studio, but we can't guarantee good support.
```
```suggestion
   **Note**: You may try newer or older versions of Android Studio, but we can't guarantee good support due to potential compatibility issues with different versions.
```

TanishMoral11 replied:
The original note warns about support but doesn’t explain why Giraffe is preferred. Adding this detail helps new contributors understand that Giraffe is the tested, reliable choice, reducing setup confusion and potential issues. It’s a small tweak that makes the guide more welcoming and clear for beginners.

adhiamboperes replied:
Added an explanation.

TanishMoral11 commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -40,20 +330,17 @@ Please follow these steps to set up Oppia Android on your local machine.
 
-2. Run the `bash scripts/setup.sh` script, which adds some development tools for Oppia Android (ktlint, checkstyle, etc.): These tools perform pre-push checks to ensure that your code is properly formatted as per Oppia standards.
+2. Run the `bash scripts/setup.sh` script, which adds some development tools for Oppia Android (ktlint, checkstyle, etc.). You should see a message indicating that the various tools have been downloaded.
 
```
```suggestion
   **Troubleshooting**: If the `setup.sh` script fails, it could be due to missing dependencies (e.g., Git or Bash) or insufficient permissions. To resolve permission issues, run `chmod +x scripts/setup.sh` to make the script executable, then try running it again. If the issue persists, ensure all required tools are installed (e.g., via `sudo apt install git` on Linux or `brew install git` on Mac) and check the terminal output for specific error messages.

```

TanishMoral11 replied:
The current text assumes the script runs perfectly, but new contributors might hit issues like permission errors or missing Git. Adding this troubleshooting note provides clear, immediate steps to fix common problems, making setup easier and less frustrating. It’s especially helpful for beginners who may not know how to debug script failures.

adhiamboperes replied:
I'm not sure about this. Git and  bash are available by default on Mac and Linux, and the windows set up steps outline how to set them both up.

I am not opposed to adding a point to the`Troubleshooting Installation` section below, once we can clarify some common causes, stacktraces and verified solutions. Do you mind openning a discussion for this?

adhiamboperes replied:
I'd like to resolve this since it is technically non-blocking. Please let us follow up in a discussion/issue.

TanishMoral11 commented on wiki/Installing-Oppia-Android.md:
```diff
@@ -5,32 +5,322 @@ This wiki page explains how to install Oppia Android on your local machine. If y
+
+Unlike Unix-based systems where Bazel runs natively without issue, the current solution on Windows is to install an Ubuntu-based subsystem. Windows currently only supports a terminal experience in this subsystem (though there is a prerelease version of the software with GUI support) which means Android Studio will not be supported. You will need to continue using the Windows version of Android St ...
+
+Due to the issues mentioned above, we recommend dual-booting your PC with Linux. However, if you choose to use Windows, please follow the instructions below:
```
```suggestion
Due to the issues mentioned above, we recommend dual-booting your PC with Linux for a smoother experience (e.g., better performance and native Bazel support). If you prefer to use Windows, please follow the instructions below:
```

adhiamboperes replied:
Changes made.

TanishMoral11 commented on wiki/Bazel-User-Guide.md:
```diff
@@ -0,0 +1,284 @@
+- [Syncing the Project](#syncing-the-project)
+
+## Overview
+Bazel is an open-source build and test tool similar to Make, Maven, and Gradle. It uses a human-readable, high-level build language.
```
```suggestion
Bazel is an open-source build and test tool similar to Make, Maven, and Gradle. It uses a human-readable, high-level build language. Oppia Android uses Bazel for its fast, scalable builds and consistent results across different environments, making it ideal for large projects like Oppia.
```

adhiamboperes replied:
Done

adhiamboperes committed:
Address reviewer comments

adhiamboperes committed:
Merge branch 'develop' of github.com:oppia/oppia-android into update-documentation

adhiamboperes commented:
Thanks @TanishMoral11, @theayushyadav11, @subhajitxyz for your reviews! I have addressed all of your comments.

You should be able to view the latest changes only by clicking on `New changes since you last viewed` near the bottom of the PR page.

Also, I have enabled auto-merge as I have self-reviewed all the chnages.

adhiamboperes committed:
Revert Feature flag
